### PR TITLE
Fix downgrade logic of `KubernetesResources` to Role `v6`

### PR DESCRIPTION
**Pull request by tigrato**
`KubernetesResources` were improperly downgraded when they grant access to all resources.

In that case, the role was downgraded to something that can't be used to access Kubernetes clusters but they could have been downgraded to a `Role` v6 with the same permissions as the `KubernetesResources`.

This commit fixes the downgrade logic to downgrade to a `Role` v6 with the same permissions as the `KubernetesResources`.

A role v7 with

```yaml
kubenretes_labels:
 '*': '*'
kubernetes_resources:
- kind: pod
  name: '*'
  namespace: '*'
  verbs:
  - '*'
```
Is downgraded to a role v6 with

```yaml
kubenretes_labels:
 '*': '*'
kubernetes_resources:
- kind: pod
  name: '*'
  namespace: '*'
```

Changelog: Fix downgrade logic of KubernetesResources to Role v6

### CODE REVIEW

**Review by AntonAM — APPROVED** [review]

File: lib/auth/grpcserver.go [hunk 1]
```diff
@@ -2224,17 +2225,41 @@ func downgradeRoleToV6(r *types.RoleV6) (*types.RoleV6, bool, error) {
...
-				types.KindKubernetesCluster,
-				types.LabelMatchers{
-					Labels: types.Labels{},
-				},
-			)
-			// Clear out the allow list so that the V6 role doesn't include unknown
-			// resources in the allow list.
-			downgraded.SetKubeResources(types.Allow, nil)
-			restricted = true
+			hasRestritiveRules := false
```
Comment (espadolini) [inline lib/auth/grpcserver.go:2228]: typo
```suggestion
			hasRestrictiveRules := false
```

File: lib/auth/grpcserver.go [hunk 2]
```diff
@@ -2224,17 +2225,41 @@ func downgradeRoleToV6(r *types.RoleV6) (*types.RoleV6, bool, error) {
...
+			for _, resource := range r.GetKubeResources(types.Allow) {
+				if resource.Kind != types.Wildcard ||
+					resource.Namespace != types.Wildcard ||
+					resource.Name != types.Wildcard ||
+					!slices.Equal(resource.Verbs, []string{types.Wildcard}) {
```
Comment (espadolini) [inline lib/auth/grpcserver.go:2233]: I'd make it more explicit here
```suggestion
					(len(resource.Verbs) != 1 || resource.Verbs[0] != types.Wildcard) {
```

Comment (espadolini) [reply lib/auth/grpcserver.go:2233]: (what if there's two verbs and they're both `"*"` tho?)

**Review by espadolini — APPROVED** [review]

File: lib/auth/grpcserver.go (hunk 2 above)
Comment (tigrato) [reply lib/auth/grpcserver.go:2233]: You can't. `CheckAndSetDefaults` prevents you from adding `*` and any other verb

**Pull request merged**